Metadata API: add Metadata.to_bytes()

MVrachev opened the pull request:
Fixes #1487 

**Description of the changes being introduced by the pull request**:

Metadata.to_bytes() is missing from the API and that is now becoming
annoying when writing the tests.
I think it makes sense to add:
it'll complete the serializing counterparts to from_bytes()/from_file().

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Metadata API: add Metadata.to_bytes()

Metadata.to_bytes() is missing from the API and that is now becoming
annoying when writing the tests.
I think it makes sense to add:
it'll complete the serializing counterparts to from_bytes()/from_file().

We can also reuse to_bytes() in to_file() and that way ensure we don't
import the JSONSerializer locally twice.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

joshuagl reviewed (commented):
Can we make use of the new `to_bytes` method from the `to_file` method? That avoids us duplicating logic which defines the default serialiser.

MVrachev commented:
I updated the pr so that `Metadata.to_file()` uses `Metadata.to_bytes()`.
Additionally, I added tests for the two cases for the `JsonSerializer` - compact and noncompact.

jku commented on tuf/api/metadata.py:
```diff
@@ -182,6 +182,20 @@ def from_bytes(
+    def to_bytes(
+        self, serializer: Optional[MetadataSerializer] = None
+    ) -> bytes:
+        """Return the bytes representation of self."""
```
Please improve this
* arguments
* raises
* I think even the return value is not yet well defined: it's not just "self as bytes", it's the serialized TUF file format as bytes)

MVrachev replied:
I updated the function docstring.

jku reviewed (approved):
LGTM, thanks
